devops/jobs/AppWatcher.groovy

```diff
@@ -61,56 +59,63 @@ class AppWatcher {
+
+                    wrappers common_wrappers
+
+                    def access_control = extraVars.get('ACCESS_CONTROL',[])
+                    access_control.each { acl ->
+                        common_read_permissions.each { perm ->
+                            authorization {
+                                permission(perm,acl)
+                            }
                         }
                     }
-                }
 
-                def APP_REPO_URL = extraVars.get("EDX_REPO_ROOT") + app_repo + "/"
+                    def APP_REPO_URL = extraVars.get("EDX_REPO_ROOT") + app_repo + "/"
 
-                // for GitHub Webhooks; note that you need to use an https URL and include
-                // the trailing slash
-                properties {
-                    if (app_repo) {
-                       githubProjectUrl(APP_REPO_URL)
+                    // for GitHub Webhooks; note that you need to use an https URL and include
+                    // the trailing slash
+                    properties {
+                        if (app_repo) {
+                           githubProjectUrl(APP_REPO_URL)
+                        }
                     }
-                }
 
-                scm {
-                    if (app_repo) {
-                        git {
-                            remote {
-                                url(APP_REPO_URL)
-                                branch(app_repo_branch)
-                            }
-                            extensions {
-                                relativeTargetDirectory('edx')
+                    scm {
+                        if (app_repo) {
+                            git {
+                                remote {
+                                    url(APP_REPO_URL)
+                                    branch(app_repo_branch)
+                                }
+                                extensions {
+                                    relativeTargetDirectory('edx')
+                                }
                             }
                         }
                     }
-                }
 
-                triggers merge_to_master_trigger(app_repo_branch)
+                    triggers merge_to_master_trigger(app_repo_branch)
 
-                steps {
-                    // trigger image-builder job, passing name of application as a parameter
-                   downstreamParameterized {
-                        trigger('image-builder') {
-                            parameters {
-                                predefinedProp('APP_NAME', app_name)
+                    steps {
+                        // trigger image-builder job, passing name of application as a parameter
+                       downstreamParameterized {
+                            trigger('../image-builders/' + app_name + '-image-builder') {
+                                parameters {
+                                    predefinedProp('APP_VERSION', '${GIT_COMMIT}')
```

Comment:
Since each application has their own builder job, it's no longer necessary to pass through the application name, since the builder job has access to that. Now we pass in the version of the application checked out by the watcher job to the builder job so it can be used in the build name.